xonsh/platform.py

```diff
@@ -0,0 +1,141 @@
+""" Module for platform-specific constants and implementations """
+
+import os
+import platform
+import sys
+
+try:
+    import distro
+except ImportError:
+    distro = None
+
+
+#
+# OS
+#
+
+ON_DARWIN = platform.system() == 'Darwin'
+ON_LINUX = platform.system() == 'Linux'
+ON_WINDOWS = platform.system() == 'Windows'
+
+ON_POSIX = (os.name == 'posix')
+
+
+
+#
+# Python & packages
+#
+
+PYTHON_VERSION_INFO = sys.version_info[:3]
+
+if PYTHON_VERSION_INFO < (3, 5, 0):
+    from pathlib import Path
+
+    class DirEntry(Path):
+        def inode(self, *args, **kwargs):
+            raise NotImplementedError
+
+    def scandir(path):
+        """ Compatibility wrapper for os.scandir from Python 3.5+ """
+        return (DirEntry(x) for x in os.listdir(path))
+else:
+    from os import scandir
+
+
+ON_ANACONDA = any(s in sys.version for s in ('Anaconda', 'Continuum'))
+
+try:
+    import pygments
+except ImportError:
+    HAS_PYGMENTS, PYGMENTS_VERSION = False, None
+else:
+    HAS_PYGMENTS, PYGMENTS_VERSION = True, pygments.__version__
+
+
+try:
+    import readline
+except ImportError:
+    HAS_READLINE = False
+else:
+    HAS_READLINE = False
+
+
+try:
+    import prompt_toolkit
```

Comment:
Hmm, I think that trying to import both readline and ptk, when normally only one will be used is pretty inefficient, especially if you happen to have both.  Please modify this to only compute these as needed.  Maybe these need to be descriptors, or be returned to functions.